app/src/main/java/com/github/multimatum_team/multimatum/model/datetime_parser/DateTimeExtractor.kt

```diff
@@ -214,20 +166,28 @@ object DateTimeExtractor {
         val alreadyProcessedTokensWithoutTimeInfo = mutableListOf<Token>()
 
         val initTokens = Tokenizer.tokenize(str)
+        markTokensFollowedByWhitespace(initTokens)
+        val nonWhiteSpaceTokens = initTokens.filter { !it.isWhitespace() }
+
         val (date, time) = recursivelyParse(
-            initTokens,
+            nonWhiteSpaceTokens,
             null,
             null,
             alreadyProcessedTokensWithoutTimeInfo
         )
         val text = alreadyProcessedTokensWithoutTimeInfo
-            .dropWhile { it is WhitespaceToken }     // ignore leading whitespaces
-            .dropLastWhile { it is WhitespaceToken } // ignore trailing whitespaces
-            .joinToString(separator = "", transform = Token::str)
-            .let(::removeMultipleWhitespaces)
+            .let(::danglingConjunctionsRemoved)
+            .joinToString(separator = "", transform = Token::strWithWhitespaceIfNeeded)
+            .removeTrailingWhitespaces()
         return DateTimeExtractionResult(text, date, time)
     }
 
+    companion object {
+        private val CONJUNCTIONS_TO_REMOVE_WHEN_DANGLING = listOf(
+            "at", "on", "next"
```

Comment:
As you say, if we want the parser to parse something else than English, deeper modifications will be needed. Moving this to strings.xml could give the misleading impression that it can parse another language without modifying anything else than resources.